### ISSUE TITLE
v5: .js configs are cached for the process lifetime due to NodeJS module caching

### ISSUE THREAD

**Issue by undeadcat** [davidtheclark/cosmiconfig#156]
Originally reported in `prettier` as https://github.com/prettier/prettier/issues/5311.
Among other changes in v5, `require-from-string` was replaced by `require` when loading .js configs [here](https://github.com/davidtheclark/cosmiconfig/commit/0143e84934d0f425af81327c12ebd1fb2eaad6c7#diff-b9cfc7f2cdf78a7f4b91a753d10865a2). `require-from-string` [had the side-effect of not caching loaded modules](https://github.com/floatdrop/require-from-string/issues/13#issuecomment-360775277). 

Without regular 'require', modules are [cached by NodeJS](https://nodejs.org/api/modules.html#modules_caching) without the possibility of clearing the cache via `clearCaches` or similar (manipulating require.cache on the client would clear it, but probably isn't something that should be delegated to clients).

This seems to be a bug because it is possible to clear the cache for .json or .yaml files.

Code sample:
Given a .prettierrc.js file:
```
module.exports = {
  tabWidth: 7
}
```
Running this:
```
var path = require('path');
var fs = require('fs');
var cosmiconfig = require('cosmiconfig');
var explorer = cosmiconfig('prettier', {sync: true});

var loadFn = explorer.loadSync || function (path) {
  return explorer.load(__filename, path)
}

var configFilePath = path.join(__dirname, '.prettierrc.js');
var fileOptions = {encoding: 'utf8'};
var previousContent = fs.readFileSync(configFilePath, fileOptions)


console.log('Before ' + loadFn(configFilePath).config.tabWidth);
//
try {
  fs.writeFileSync(configFilePath, 'module.exports = {tabWidth:2}', fileOptions)
  explorer.clearCaches();
  console.log('After: ' + loadFn(configFilePath).config.tabWidth);

} finally {
  //restore previous content just to make this script repeatable locally.
  fs.writeFileSync(configFilePath, previousContent, fileOptions);
}
```
With `cosmiconfig` 3.1.0 this produces:
```
Before 7
After: 2
```
With 5.0.6: 
```
Before 7
After: 7
```

**Comment by olsonpm** [comment]
Hey thanks for the details.  I should have time this week to look into it.

**Comment by olsonpm** [comment]
@oomathias - cosmiconfig has its own cache.  The problem is the builtin `require` means an additional cache is being used.

And as much as I don't like mucking with global state, deleting the matching `require.cache` entries seems to be the only sensible solution assuming we still want to use `require`.

**Comment by davidtheclark** [comment]
Fixed by #158. Published as 5.0.7. Thanks @olsonpm!
